algorithm-solution/src/main/java/problemset/TwoSumWithDuplicatesStableSolution.java

```diff
@@ -0,0 +1,44 @@
+/*
+Problem : TwoSum
+Given an array of integers, return indices of the two numbers such that they add up to a specific target.
+You may assume that each input would have exactly one solution, and you may not use the same element twice.
+Example:
+Given nums = [2, 7, 11, 15], target = 9,
+Because nums[0] + nums[1] = 2 + 7 = 9,
+return [0, 1]
+*/
+
+class TwoSumWithDuplicatesStableSolution {
+    public int[] twoSum(int[] nums, int target) {
+        Map<Integer, List<Integer>> pos = new HashMap<>();
+        
+        for(int i = 0; i<nums.length; i++) {
+            List<Integer> posL = pos.get(nums[i]);
+            if( posL == null) {
+                posL = new ArrayList<>();
+                pos.put(nums[i], posL);
+            } 
+            posL.add(i);
+        }
+        
+        
+        Integer pairPos = null;
+        for(int i = 0; i<nums.length; i++) {
+            List<Integer> tPosL = pos.get(target-nums[i]);
+            if(tPosL != null) {
+                for(Integer tpos : tPosL) {
+                    if(tpos != i) {
+                        pairPos = tpos;
+                        break;
+                    }    
+                }
+            }
+            
+            if(pairPos != null) {
+                return new int[]{i, (int)pairPos};
+            }
+        }
+        return null;
+    }
+        
```

Comment:
I Liked your solution. It's way better in execution time.
I did little improvements. Just to make decisions while iterating over array itself.
It reduces execution time to half in leetcode IDE.

`public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> elementIndexMap = new HashMap<>();
        for (int i = 0; i <= nums.length - 1; i++) {
            //If ELEMENT index exists
            if (elementIndexMap.get(nums[i]) == null) {
                // If COMPLEMENT OF ELEMENT's index exists
                if ((elementIndexMap.get(target - nums[i]) != null)) {
                    return new int[]{elementIndexMap.get(target - nums[i]), i};
                }
                elementIndexMap.put(nums[i], i);
            } // if element's index exists and element is half of target
            else if (2 * nums[i] == target) {
                return new int[]{elementIndexMap.get(nums[i]), i};
            }
        }
        return null;
    } `

Reply:
Wow.. It is a very simplified solution! I liked it..

Reply:
Can make the changes in the solution. lets also list the solutions in the  ` README.md ` ?

Reply:
shall we find the best solution in term of execution in leetcode if possible?

Reply:
I have further simplified the solution as follows, you do you think?
`class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pos = new HashMap<>();
        
        for(int i = 0; i<nums.length; i++) {
            if( pos.get(target-nums[i]) != null) {
              return new int[]{pos.get(target-nums[i]), i};  
            } else {
              pos.put(nums[i], i);
            }
        }
     return null;
    }
        
}`

Reply:
Updated the code, check once?

Reply:
Yes We should get teh best solution every tiem and then merge it :+1: